File: 프로그래머스/1/135808. 과일 장수/과일 장수.py
def solution(k, m, score):
    answer = 0
    box = []
    score.sort(reverse=True)
    
    for i in range(len(score) // m):
        answer += score[m*i:m*(i+1)][-1] * m
    return answer

print(solution(3, 4, [1, 2, 3, 1, 2, 3, 1]))

File: 프로그래머스/1/135808. 과일 장수/과일 장수.java
import java.util.*;

class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        
        // 오름차순 정렬
        Arrays.sort(score);
        
        // 내림차순 정렬
        for(int i=0; i<score.length / 2; i++) {
            int temp = score[i];
            score[i] = score[score.length-1-i];
            score[score.length-1-i] = temp;
        }
        
        for(int i=0; i<score.length/m; i++) {
            int[] arr = Arrays.copyOfRange(score, m*i, m*(i+1));
            answer += arr[arr.length - 1] * m;
        }
        return answer;
    }
}